src/Polly.Core/Strategy/OutcomePredicate.TResult.cs

```diff
@@ -0,0 +1,222 @@
+using System;
+using Polly.Strategy;
+
+namespace Polly.Strategy;
+
+/// <summary>
+/// Predicate that uses <see cref="Outcome{TResult}"/> and <typeparamref name="TResult"/> as an input.
+/// </summary>
+/// <typeparam name="TArgs">The type of arguments the predicate uses.</typeparam>
+/// <typeparam name="TResult">The result type that this predicate handles.</typeparam>
+public sealed class OutcomePredicate<TArgs, TResult>
+    where TArgs : IResilienceArguments
+{
+    private readonly List<Func<Outcome<TResult>, TArgs, ValueTask<bool>>> _predicates = new();
+
+    /// <summary>
+    /// Gets a value indicating whether the predicate is empty.
+    /// </summary>
+    public bool IsEmpty => _predicates.Count == 0;
+
+    /// <summary>
+    /// Adds an exception predicate for the specified exception type.
+    /// </summary>
+    /// <typeparam name="TException">The exception type to add a predicate for.</typeparam>
+    /// <returns>The current updated instance.</returns>
+    public OutcomePredicate<TArgs, TResult> HandleException<TException>()
+        where TException : Exception
+    {
+        return HandleOutcome((outcome, _) =>
+        {
+            if (outcome.Exception is TException typedException)
```

Comment:
We don't seem to use/need the actual value?

```suggestion
            if (outcome.Exception is TException)
```

Reply:
Done.